### PR TITLE
doc(website): Adding how-to guide for using Renovate

### PR DESCRIPTION
**Pull request by cloudmatt**
#### Summary

Adding a `how-to` guide for leveraging [Renovate](https://docs.renovatebot.com/) to keep CloudQuery plugins up-to-date per https://github.com/cloudquery/cloudquery/issues/6335. I did follow the setup instructions and ran the website locally and everything appears to load fine on my end the only exception being the header which renders a weird space:

image

but that's due to the `team.ts` content being loaded that populates that with the avatar/links for CloudQuery employees.

### CODE REVIEW

**Comment by cloudmatt** [comment]
I don't know how to resolve it but it looks like vale is refusing to acknowledge my inline off/on [here](https://github.com/cloudquery/cloudquery/pull/6435/files#diff-11a859327ea29ed13104c9c37e306c1970d7058d7019f3ff1c468bbab0b582f8R27-R36) and is failing it based on the legitimate spelling of those words. Any tips to make value happier?

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+### Step 2: **Create Custom RegexManager**
+
+Now that we've identified the fields we need to extract from the configuration file to create the custom regex, next we need to construct the regex statement to extract those 'keys' into named capture groups.
+
+For this it's recommended to use a regex testing tool to ensure your regex is actually working. [Regex101](https://regex101.com/) is a popular one but feel free to use any others as long as they support the PCRE2 regex engine. So taking our sample configuration file from above, we begin to play around with various regex until we meet the following criteria:
+
+- a named capture group for the plugin type (source or destination) which we will reference as 'kind' in this tutorial.
+- a named capture group for the plugin name (aws, gcp, etc..) which we will reference as 'plugin' in this tutorial.
+- a named capture group for the plugin current version which **must** be named currentValue.
```
Comment (hermanschaaf) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:79]:
```suggestion
- a named capture group for the plugin current version which **must** be named `currentValue`.
```

Putting this inside backticks should keep the linter happy

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+```shell copy
+kind:\s(?<kind>.*)\nspec:\n\s{2}name:.*?\w\n\s{2}path:\scloudquery\/(?<plugin>.*)\n\s{2}version:\s\"?v(?<currentValue>.*\d)\"?\n
+```
+This regular expression does the following:
+- captures the `kind: (source|destination)` into a named capture group called 'kind`
+- skips the `spec:` and `name:` lines of the file
+- captures the `path: cloudqery/(gcp, aws, etc...)` into a named capture group called `plugin` while escaping the `cloudquery/` prefix.
+{/*<!-- vale off -->*/}
+- skips to the next line and captures the `version: "vX.X.X" into the **required** named capture group 'currentValue' while optionally checking if the string is quoted or not.
```
Comment (hermanschaaf) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:91]:
```suggestion
- skips to the next line and captures the `version: "vX.X.X"` into the **required** named capture group `currentValue` while optionally checking if the string is quoted or not.
```

File: .github/styles/Vocab/Base/accept.txt [hunk 3]
```diff
@@ -153,6 +160,7 @@ upserted
 uuid
 UUID
 Vercel
+versioningTemplate
```
Comment (hermanschaaf) [inline .github/styles/Vocab/Base/accept.txt:163]: I think we should be able to remove these exceptions if we place the words inside backticks in the post

File: website/pages/how-to-guides/update-plugins-using-renovate.md (hunk 1 above)
Comment (cloudmatt) [reply website/pages/how-to-guides/update-plugins-using-renovate.md:79]: sorry - just saw this after my last commit, I added them to the `.github/styles/proselint/Vocab/Base/accept.txt` file. Want me to back that commit out and backtick them instead?

**Comment by cloudmatt** [comment]
```shell
Getting links from: https://cloudquery-9g650z65a-cloudquery.vercel.app/how-to-guides/update-plugins-using-renovate
├─BROKEN─ https://www.mend.io/free-developer-tools/renovate/ (HTTP_403)
├─BROKEN─ https://www.mend.io/free-developer-tools/renovate/ (HTTP_403)
Finished! 269 links found. 2 broken.
```
no idea what that's checking - but those links do work. What should I change them to?

**Comment by hermanschaaf** [comment]
@cloudmatt Huh, weird. Maybe they block certain IP ranges, making our automated check fail? Maybe try update the link to https://github.com/marketplace/renovate?

**Comment by cloudmatt** [comment]
> @cloudmatt Huh, weird. Maybe they block certain IP ranges, making our automated check fail? Maybe try update the link to https://github.com/marketplace/renovate?

I'll just update those links and point directly to their github page - I don't necessarily want to point to the github app because renovate supports other SVNs (GitLab, BitBucket, etc..) but I think just pointing to their code repo in github is a good middle ground.

**Comment by hermanschaaf** [comment]
@cloudmatt I added an entry to `team.ts` for you, as well as a few other small fixes. Feel free to update that team.ts entry with more / different details if you like. Overall looks great! 🎉

**Review by hermanschaaf — APPROVED** [review]

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+While Renovate provides tons of built-in functionality to make most dependency management simple, it also provides a powerful functionality called 'RegexManager' which allows you to leverage custom regular expression statements to manually extract dependencies, their version, and updating mechanism. Click [here](https://docs.renovatebot.com/modules/manager/regex/) for more details on the specific configurations of RegexManager.
+
+For our purposes, we're concerned about the following items:
+- `datasourceTemplate`: datasource is where the package's versions are managed.
+- `packageNameTemplate`:  this is the lookup name that is passed to the datasource.
+- `depNameTemplate`: this is the 'friendly' name that is shown in renovate PRs.
+- `extractVersionTemplate`: regular expression for discovering plugin version in the datasource.
+- `versioningTemplate`: what versioning 'format' the datasource follows
+- `fileMatch`: path to the cloudquery.yaml file in your repository.
```
Comment (erezrokah) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:34]:
```suggestion
- `fileMatch`: path to the `cloudquery.yaml` file in your repository.
```

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+```yaml copy
+---
+kind: source
+spec:
+  name: gcp
+  path: cloudquery/gcp
+  version: "v5.4.0"
+  tables: ["*"]
+  sestinations: ["postgresql"]
```
Comment (erezrokah) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:47]:
```suggestion
  destinations: ["postgresql"]
```

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+Now that we've identified the fields we need to extract from the configuration file to create the custom regex, next we need to construct the regex statement to extract those 'keys' into named capture groups.
+
+For this it's recommended to use a regex testing tool to ensure your regex is actually working. [Regex101](https://regex101.com/) is a popular one but feel free to use any others as long as they support the PCRE2 regex engine. So taking our sample configuration file from above, we begin to play around with various regex until we meet the following criteria:
+
+- a named capture group for the plugin type (source or destination) which we will reference as 'kind' in this tutorial.
+- a named capture group for the plugin name (aws, gcp, etc..) which we will reference as 'plugin' in this tutorial.
+- a named capture group for the plugin current version which **must** be named `currentValue`.
+
+For our sample configuration file this regex would look something like:
```
Comment (erezrokah) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:81]:
```suggestion
For our sample configuration file this regex will look like:
```

File: website/pages/how-to-guides/update-plugins-using-renovate.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+- skips the `spec:` and `name:` lines of the file
+- captures the `path: cloudqery/(gcp, aws, etc...)` into a named capture group called `plugin` while escaping the `cloudquery/` prefix.
+- skips to the next line and captures the `version: "vX.X.X"` into the **required** named capture group `currentValue` while optionally checking if the string is quoted or not.
+
+
+On the Regex101 site we can verify the regex is working by viewing the explanation section on the left side of the screen:
+
+![screenshot of regex101](/images/blog/manage-plugins-with-renovate/renovate-howto-regex101.png)
+
+**Important Note**: This regex is _specific_ to the example shown in this how-to guide, if your configuration file is formatted different (such as lines in different orders or indented differently like in a helm values.yaml file) then you'll need to adjust the regex accordingly.
```
Comment (erezrokah) [inline website/pages/how-to-guides/update-plugins-using-renovate.md:97]:
```suggestion
**Important Note**: This regex is _specific_ to the example shown in this how-to guide, if your configuration file is formatted different (such as lines in different orders or indented differently like in a helm `values.yaml` file) then you'll need to adjust the regex accordingly.
```

**Review by erezrokah — APPROVED** [review]
Looks great, I fixed a few minor issues, but let's ship this.
If we need we can follow up in another PR.

Thanks for the quick turn around @cloudmatt 🚀

**Pull request merged**